doc/examples/realtime_servo/realtime_servo_tutorial.rst

```diff
@@ -1,181 +1,196 @@
+
+In addition to the servoing capability, MoveIt Servo has some convenient features such as:
+
+    - Checking for singularities
+    - Checking for collisions
+
+The collision checking is an optional feature that can be disabled using the ``check_collisions`` parameter.
+
+Using the C++ API
+------------------
+The C++ interface allows the user to create and utilize an instance of Servo within their C++ programs.
+This can be beneficial when there is a performance requirement to avoid the overhead of ROS communication infrastucture,
+or when the output generated by Servo needs to be fed into some other controller that does not have a ROS interface.
+
+When using Servo with the C++ interface the three input command types are ``JointJogCommand``, ``TwistCommand`` and ``PoseCommand``.
+The output from Servo when using the C++ interface is ``KinematicState``, a struct containing joint names, positions, velocities and accelerations.
+As given by the definitions in `datatypes <https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/include/moveit_servo/utils/datatypes.hpp>`_ header file.
+
+The first step is to create a Servo instance.
+
+.. code-block:: c++
+
+    // Import the Servo headers.
+    #include <moveit_servo/servo.hpp>
+    #include <moveit_servo/utils/common.hpp>
+
+.. code-block:: c++
 
-To run the demo, make sure your controller is plugged in and can be detected by :code:`ros2 run joy joy_node`. Usually this happens automatically after plugging the controller in. Then launch with ::
+    // The node to be used by Servo.
+    rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("servo_tutorial");
 
-    ros2 launch moveit_servo servo_example.launch.py
+.. code-block:: c++
 
-Make a service request to start Servo ::
+    // Get the Servo parameters.
+    const std::string param_namespace = "moveit_servo";
+    const std::shared_ptr<const servo::ParamListener> servo_param_listener =
+        std::make_shared<const servo::ParamListener>(node, param_namespace);
+    const servo::Params servo_params = servo_param_listener->get_params();
 
-    ros2 service call /servo_node/start_servo std_srvs/srv/Trigger {}
+.. code-block:: c++
 
-You should be able to control the arm with your controller now, with MoveIt Servo automatically avoiding singularities and collisions.
+    // Create the planning scene monitor.
+    const planning_scene_monitor::PlanningSceneMonitorPtr planning_scene_monitor =
+        createPlanningSceneMonitor(node, servo_params);
 
-Without a Controller
-^^^^^^^^^^^^^^^^^^^^
+.. code-block:: c++
 
-If you do not have a joystick or game controller, you can still try the demo using your keyboard. With the demo still running, in a new terminal, run ::
+    // Create a Servo instance.
+    Servo servo = Servo(node, servo_param_listener, planning_scene_monitor);
```

Comment:
Either use a single code block or add some non-code text with explanations in between these.

Reply:
Did not add explanations between  the blocks since I put comments in the blocks, making it a single block with inline comments might be the better idea.